sdk/src/main/scala/io/horizen/account/state/McAddrOwnershipMsgProcessor.scala

```diff
@@ -243,6 +202,98 @@ case class McAddrOwnershipMsgProcessor(params: NetworkParams) extends NativeSmar
     newOwnershipId
   }
 
+
+  def doAddNewMultisigOwnershipCmd(invocation: Invocation, view: BaseAccountStateView, msg: Message): Array[Byte] = {
+
+    // check that message contains a nonce, in the context of RPC calls the nonce might be missing
+    if (msg.getNonce == null) {
+      val errMsg = s"Call must include a nonce: msg = $msg"
+      log.warn(errMsg)
+      throw new ExecutionRevertedException(errMsg)
+    }
+
+    // check that invocation.value is zero
+    if (invocation.value.signum() != 0) {
+      val errMsg = s"Value must be zero: invocation = $invocation"
+      log.warn(errMsg)
+      throw new ExecutionRevertedException(errMsg)
+    }
+
+    val senderScAddress = msg.getFrom
+
+    // check that sender account exists
+    if (!view.accountExists(senderScAddress) ) {
+      val errMsg = s"Sender account does not exist: msg = $msg"
+      log.warn(errMsg)
+      throw new ExecutionRevertedException(errMsg)
+    }
+
+    val inputParams = getArgumentsFromData(invocation.input)
+
+    val cmdInput = AddNewMultisigOwnershipCmdInputDecoder.decode(inputParams)
+    val mcMultisigAddress = cmdInput.mcTransparentAddress
+    val mcSignatures = cmdInput.mcSignatures
+    val redeemScript = cmdInput.redeemScript
+
+    // get threshold signature value and all pub keys from redeemScript.
+    // If any semantic error is detected while parsing an exception is raised
+    var retValue : (Int, Seq[Array[Byte]]) = null
+    try {
+      retValue = checkMcRedeemScriptForMultisig(redeemScript)
+    } catch {
+      case e: IllegalArgumentException =>
+        val errMsg = s"Unexpected format of redeemScript: ${e.getMessage}"
+        log.warn(errMsg)
+        throw new ExecutionRevertedException(errMsg)
+    }
+    val thresholdSignatureValue = retValue._1
+    val pubKeys = retValue._2
+
+    // check validity of the addr/redeemscript pair
+    if(!checkMultisigAddress(mcMultisigAddress, redeemScript, networkParams)) {
```

Comment:
Maybe it is better to first check that the multisigAddress corresponds to the redeemScript (or vice-versa) and then calling checkMcRedeemScriptForMultisig. I suppose that an invalid redeemScript won't ever verify a multisigAddress, while a valid redeemScript could still not verifying the mcMultisigAddress.
In addition, once we have the pubkeys from the redeemScript, we could check that the number of signatures could not be greater that the number of pubkeys.